### PR TITLE
Add openbsd ports manifest support

### PR DESCRIPTION
**Pull request by blackgnezdo**

### CODE REVIEW

**Comment by blackgnezdo** [comment]
I'm not suggesting this is merged as is, more asking for feedback and guidance about which parts I should documented.

**Review by phadej — CHANGES_REQUESTED** [review]
Please, don't use `ormolu`.

I won't look at the patch if it is formatted greatly differently then the rest of the code base.

**Comment by blackgnezdo** [comment]
> Please, don't use `ormolu`.
> 
> I won't look at the patch if it is formatted greatly differently then the rest of the code base.

Sorry about that, could you check if the second attempt at reformatting in commit 2 is more agreeable? 

If not I'll try to make a manual formatting pass on top of commit 1. I just was hacking it without regard for formatting in hope I can defer formatting to a tool...

**Comment by phadej** [comment]
The four spaces indentation (with `where` half-indented with two spaces) is my "requirement". The non-configurability of ormolu is why I don't use it. See https://github.com/tibbe/haskell-style-guide/blob/master/haskell-style.md (my style differes in details, but I'm lenient about the small differences).

**Comment by blackgnezdo** [comment]
> Please, don't use `ormolu`.
> 
> I won't look at the patch if it is formatted greatly differently then the rest of the code base.

Done.

**Comment by blackgnezdo** [comment]
Any feedback?

**Comment by phadej** [comment]
I haven't forgotten about this.

File: cabal-bundler/src/CabalBundler/Main.hs [hunk 1]
```diff
@@ -100,6 +102,7 @@ optsP = Opts
     <*> optional (O.option (O.eitherReader $ return . fromFilePath) (O.short 'p' <> O.long "plan" <> O.metavar "PATH" <> O.help "Use plan.json provided"))
 
 formatP :: O.Parser Format
-formatP = nixSingle <|> curl <|> pure Curl where
+formatP = nixSingle <|> curl <|> openbsd <|> pure Curl where
     nixSingle = O.flag' NixSingle $ O.long "nix-single" <> O.help "Single nix derivation"
     curl      = O.flag' Curl      $ O.long "curl"       <> O.help "Curl script to download dependencies"
+    openbsd   = O.flag' OpenBSD   $ O.long "openbsd"    <> O.help "OpenBSD port manifest"
```
Comment (phadej) [inline cabal-bundler/src/CabalBundler/Main.hs:108]: Is 

```
	Cabal	3.0.0.0	0	\
	ansi-terminal	0.10.1	0	\
	ansi-wl-pprint	0.6.9	2	\
	cabal-fmt	0.1.1	0	\
	colour	2.3.5	0	\
	optparse-applicative	0.15.1.0	1	\
	transformers-compat	0.6.5	0	\
```

the **whole** port manifest file, or is it just a fragment?

File: cabal-bundler/src/CabalBundler/OpenBSD.hs [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+        ver = prettyShow $ depVersion dep
+        rev = show $ depRevision dep
+     in intercalate "\t" ["", name, ver, rev, "\\"]
+
+findExe :: String -> Map P.UnitId P.Unit -> [P.UnitId]
+findExe exeName units =
+    [ uid
+    | (uid, unit) <- M.toList units
+    , (P.CompNameExe e, _) <- M.toList (P.uComps unit)
+    , Just (P.LocalUnpackedPackage _) <- [P.uPkgSrc unit]
```
Comment (phadej) [inline cabal-bundler/src/CabalBundler/OpenBSD.hs:131]: Why is this required?

In my branch (adapting to recent changes in framework) to take `packageName` and filter based on it.

```haskell
findExe :: PackageName ->  String -> Map P.UnitId P.Unit -> [P.UnitId]
findExe pn exeName units =
    [ uid
    | (uid, unit) <- M.toList units
    , pkgName (toCabal (P.uPId unit)) == pn
    , (P.CompNameExe e, _) <- M.toList (P.uComps unit)
    , e == T.pack exeName
    ]
```

File: cabal-bundler/src/CabalBundler/Main.hs (hunk 1 above)
Comment (phadej) [reply cabal-bundler/src/CabalBundler/Main.hs:108]: Relatedly, is there a page explaining what "port manifest" files look like. I'd like to add acomment to the top of the `OpenBSD` module, but I have hard time finding any information.

(Don't worry too much about that, it's e.g. impossible to find anything concrete about nix derivations as well).

**Review by phadej — COMMENTED** [review]
This looks OK. I have a branch adopting this to recent changes and adding fixture tests. Will put it up once #34 is merged.

File: cabal-bundler/src/CabalBundler/Main.hs (hunk 1 above)
Comment (blackgnezdo) [reply cabal-bundler/src/CabalBundler/Main.hs:108]: This is a fragment of a port Makefile, the value for MODCABAL_MANIFEST variable. 

I will eventually have a stable reference for where this is documented, but I don't have a predictable location yet.

File: cabal-bundler/src/CabalBundler/OpenBSD.hs (hunk 2 above)
Comment (blackgnezdo) [reply cabal-bundler/src/CabalBundler/OpenBSD.hs:131]: You may recall that this code was written in a different context (cabal-bootstrap-gen). I may have missed a better way to do it as I moved over to cabal-bundler. Maybe point me at your branch so I can test it?

Comment (blackgnezdo) [reply cabal-bundler/src/CabalBundler/OpenBSD.hs:131]: I found https://github.com/phadej/cabal-extras/tree/pr-32-openbsd-ports-manifest, will use that.

File: cabal-bundler/src/CabalBundler/Main.hs (hunk 1 above)
Comment (phadej) [reply cabal-bundler/src/CabalBundler/Main.hs:108]: Thanks, I amended the `--help` message to mention `MODCABAL_MANIFEST` variable.

Could you check #35 that I didn't break anything and specifically that the golden fixture looks correct

File: cabal-bundler/src/CabalBundler/OpenBSD.hs (hunk 2 above)
Comment (phadej) [reply cabal-bundler/src/CabalBundler/OpenBSD.hs:131]: Yes, that the one to look at. I just forcepushed the update there though.

**Comment by phadej** [comment]
Merged in #35
